cmd/clusterctl/pkg/client/common.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/client/repository"
+)
+
+// getComponentsByName is a utility methods that returns provider component for a given provider, targetNamespace and WatchingNamespace.
+func (c *clusterctlClient) getComponentsByName(provider string, targetNamespace string, watchingNamespace string) (repository.Components, error) {
+
+	// parse the abbreviated syntax for name[:version]
+	name, version, err := parseProviderName(provider)
+	if err != nil {
+		return nil, err
+	}
+
+	// if a target namespace is defined, use it, otherwise let the namespace empty so the default one defined in the
+	// provider components YAML will be used
+	var namespace string
+	if targetNamespace != "" {
+		namespace = targetNamespace
+	}
+
+	// gets the provider configuration (that includes the location of the provider repository)
+	providerConfig, err := c.configClient.Providers().Get(name)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Do you want to add a nil-check for providerConfig here, just to be safe?

Reply:
Get return an object if the selected provider exists, or not in case it does not exists.
 This makes the nil check not necessary